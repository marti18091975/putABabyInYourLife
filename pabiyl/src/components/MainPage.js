import React, { useEffect, useState } from 'react';
import './mainPage.css';
import { NavLink } from 'react-router-dom';
import { login } from '../actions/AuthActions';
import authStore from '../stores/authStore';
import { loadDetails } from "../actions/detailAction";
import userStore from "../stores/userStore";
function MainPage(props) {
    const [userEmail, setUserEmail] = useState(authStore.getUserEmail());
    useEffect(() => {
        authStore.addChangeListener(onChange);
        setUserEmail(authStore.getUserEmail());
        console.log("AQUÏ TENIM LEMAIL", userEmail);

        return () => authStore.removeChangeListener(onChange);
    }, []);

    function onChange() {
        setUserEmail(authStore.getUserEmail());
    }

    return (

        <main className="main__main">
            <div className="title__main">Put a baby in your life</div>
            <div className="navigation__container">
                <div className="navigation__container--side">
                    <NavLink to="/profile" className="navigation__text">profile</NavLink>
                    <NavLink to="/SearchFilters/" className="navigation__text" >find users</NavLink>
                    <NavLink to="/detailUser/5f4e6fc673d494545cfbadfc" className="navigation__text">messages</NavLink>
                </div>
                <div className="navigation__container--side">
                    <NavLink to="/detailUser/5f4e6fc673d494545cfbadfc" className="navigation__text">group activities</NavLink>
                    <NavLink to="/detailUser/5f4e6fc673d494545cfbadfc" className="navigation__text">chat</NavLink>
                </div>
                <div className="navigation__container--side">
                    <NavLink to="/detailUser/5f4e6fc673d494545cfbadfc" className="navigation__text">users experiencies</NavLink>
                    <NavLink to="/detailUser/5f4e6fc673d494545cfbadfc" className="navigation__text">legal doubts</NavLink>
                </div>

            </div>
        </main >
    )
}
export default MainPage;
