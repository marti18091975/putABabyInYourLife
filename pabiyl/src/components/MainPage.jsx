import React from "react";
import "./mainPage.scss";
import { NavLink } from "react-router-dom";

function MainPage(props) {
  return (
    <main className="main__main">
      <div className="title__main">Put a baby in your life</div>
      <div className="navigation__container">
        <NavLink to="/profile" className="navigation__text">
          PROFILE
        </NavLink>
        <NavLink to="/searchFilters/" className="navigation__text">
          FIND USERS
        </NavLink>
        <NavLink to="/messages" className="navigation__text">
          MESSAGES
        </NavLink>
        <section className="main__footer">
          <img
            src="https://media-exp1.licdn.com/dms/image/C5603AQEZPiAr7_RCxA/profile-displayphoto-shrink_400_400/0?e=1602115200&v=beta&t=z8RyBtkp3SCyRZEnFBW1KU286hrCLlbxDYxbfSLyen8"
            className="image__footer"
            alt=""
          ></img>
          <div className="text__footer">Mart√≠ Amat Vila</div>

          <img
            src="https://www.skylabcoders.com/images/403/default.png"
            className="icon__footer"
            alt=""
          ></img>
        </section>
      </div>
    </main>
  );
}
export default MainPage;
