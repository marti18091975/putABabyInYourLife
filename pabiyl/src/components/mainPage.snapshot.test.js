import renderer from 'react-test-renderer';
import React from 'react';
import MainPage from './MainPage';
import { BrowserRouter as Router } from 'react-router-dom';

describe('App snapshot', () => {

    const tree = renderer.create(
        <Router>
            <MainPage />
        </Router>
    );

    it('should match', () => {
        expect(tree.toJSON()).toMatchSnapshot();
    });
});