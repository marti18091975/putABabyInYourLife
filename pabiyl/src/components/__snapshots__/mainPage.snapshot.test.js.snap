// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App snapshot should match 1`] = `
<main
  className="main__main"
>
  <div
    className="title__main"
  >
    Put a baby in your life
  </div>
  <div
    className="navigation__container"
  >
    <a
      aria-current={null}
      className="navigation__text"
      href="/profile"
      onClick={[Function]}
    >
      PROFILE
    </a>
    <a
      aria-current={null}
      className="navigation__text"
      href="/searchFilters/"
      onClick={[Function]}
    >
      FIND USERS
    </a>
    <a
      aria-current={null}
      className="navigation__text"
      href="/messages"
      onClick={[Function]}
    >
      MESSAGES
    </a>
    <section
      className="main__footer"
    >
      <img
        alt=""
        className="image__footer"
        src="https://media-exp1.licdn.com/dms/image/C5603AQEZPiAr7_RCxA/profile-displayphoto-shrink_400_400/0?e=1602115200&v=beta&t=z8RyBtkp3SCyRZEnFBW1KU286hrCLlbxDYxbfSLyen8"
      />
      <div
        className="text__footer"
      >
        Mart√≠ Amat Vila
      </div>
      <img
        alt=""
        className="icon__footer"
        src="https://www.skylabcoders.com/images/403/default.png"
      />
    </section>
  </div>
</main>
`;
